cobigen-eclipse/cobigen-eclipse/src/com/devonfw/cobigen/eclipse/common/tools/ResourcesPluginUtil.java

```diff
@@ -105,43 +106,18 @@ public static void refreshConfigurationProject() {
    */
   public static IProject getGeneratorConfigurationProject() throws GeneratorProjectNotExistentException, CoreException {
 
-    File templatesDirectory = getTemplatesDirectory();
-
-    generatorProj = ResourcesPlugin.getWorkspace().getRoot().getProject(ResourceConstants.CONFIG_PROJECT_NAME);
-
-    if (!generatorProj.exists()) {
-      if (!isUpdateDialogShown) {
-        if (templatesDirectory.exists()) {
-          Path jarFilePath = TemplatesJarUtil.getJarFile(false, templatesDirectory.toPath());
-          // If we don't find at least one jar, then we do need to download new templates
-          if (jarFilePath == null || !Files.exists(jarFilePath)) {
-            int result = createUpdateTemplatesDialog();
-            isUpdateDialogShown = true;
-            if (result == 1) {
-              // User does not want to download templates.
-              userWantsToDownloadTemplates = false;
-            } else {
-              userWantsToDownloadTemplates = true;
-            }
-          }
-
-        } else {
-          int result = createUpdateTemplatesDialog();
-          isUpdateDialogShown = true;
-          if (result == 1) {
-            // User does not want to download templates.
-            userWantsToDownloadTemplates = false;
-          } else {
-            userWantsToDownloadTemplates = true;
-          }
-        }
-      }
-    }
-    if (userWantsToDownloadTemplates) {
+    generatorProj = ResourcesPlugin.getWorkspace().getRoot()
+        .getProject(ResourceConstants.TEMPLATE_SETS_CONFIG_PROJECT_NAME);
+    if (generatorProj.exists()) {
       return generatorProj;
     } else {
-      return null;
+      generatorProj = ResourcesPlugin.getWorkspace().getRoot().getProject(ResourceConstants.CONFIG_PROJECT_NAME);
+      if (generatorProj.exists()) {
+        return generatorProj;
```

Comment:
Add comments to show that here two cases are being checked (template sets and monolithic templates)